### PR TITLE
[Refactoring] Limit arguments of IMAP processor

### PR DESCRIPTION
**Pull request by chibenwa**
Tag and Command can be obtained from an IMAP request

### CODE REVIEW

File: protocols/imap/src/main/java/org/apache/james/imap/api/Tag.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    private final String value;
+
+    public Tag(String value) {
+        Preconditions.checkNotNull(value);
+        this.value = value;
+    }
+
+    public String getValue() {
+        return value;
+    }
```
Comment (rouazana) [inline protocols/imap/src/main/java/org/apache/james/imap/api/Tag.java]: missing toString / equals? + equalsVerifier?

File: protocols/imap/src/main/java/org/apache/james/imap/processor/AbstractAuthProcessor.java [hunk 2]
```diff
@@ -104,23 +102,23 @@ protected void doAuthWithDelegation(AuthenticationAttempt authenticationAttempt,
...
                 }
             }
             if (authFailure) {
-                manageFailureCount(session, tag, command, responder, failed);
+                manageFailureCount(session, request, responder, failed);
             }
         } catch (UserDoesNotExistException e) {
             LOGGER.info("User {} does not exist", authenticationAttempt.getAuthenticationId(), e);
-            no(command, tag, responder, HumanReadableText.USER_DOES_NOT_EXIST);
+            no(request, responder, HumanReadableText.USER_DOES_NOT_EXIST);
```
Comment (rouazana) [inline protocols/imap/src/main/java/org/apache/james/imap/processor/AbstractAuthProcessor.java]: how can you answer "no" without the tag?

File: protocols/imap/src/main/java/org/apache/james/imap/processor/AbstractMailboxProcessor.java [hunk 3]
```diff
@@ -334,28 +334,28 @@ private void handleResponseException(ImapProcessor.Responder responder, MailboxE
...
-        final StatusResponse response = factory.taggedOk(tag, command, HumanReadableText.COMPLETED);
+    protected void okComplete(ImapRequest request, ImapProcessor.Responder responder) {
+        final StatusResponse response = factory.taggedOk(request.getTag(), request.getCommand(), HumanReadableText.COMPLETED);
         responder.respond(response);
     }
 
-    protected void okComplete(ImapCommand command, Tag tag, ResponseCode code, ImapProcessor.Responder responder) {
-        final StatusResponse response = factory.taggedOk(tag, command, HumanReadableText.COMPLETED, code);
+    protected void okComplete(ImapRequest request,  ResponseCode code, ImapProcessor.Responder responder) {
+        final StatusResponse response = factory.taggedOk(request.getTag(), request.getCommand(), HumanReadableText.COMPLETED, code);
```
Comment (rouazana) [inline protocols/imap/src/main/java/org/apache/james/imap/processor/AbstractMailboxProcessor.java]: oh I see... you are getting it from the request.
It is always right? I believe is some cases (unsolicited response) the tag could be different than the request's one, no?

File: protocols/imap/src/main/java/org/apache/james/imap/processor/AbstractAuthProcessor.java (hunk 2 above)
Comment (chibenwa) [reply protocols/imap/src/main/java/org/apache/james/imap/processor/AbstractAuthProcessor.java]: by calling the responder+response factory. That's orthogonal to this refactoring.

File: protocols/imap/src/main/java/org/apache/james/imap/processor/AbstractMailboxProcessor.java [hunk 4]
```diff
@@ -355,7 +355,7 @@ protected void bye(ImapProcessor.Responder responder, HumanReadableText key) {
         responder.respond(response);
     }
 
-    protected abstract void processMessage(M message, ImapSession session, Responder responder);
+    protected abstract void processRequest(M message, ImapSession session, Responder responder);
```
Comment (remk) [inline protocols/imap/src/main/java/org/apache/james/imap/processor/AbstractMailboxProcessor.java]: Can you rename the parameter name too ?

**Comment by Arsnael** [comment]
Rebase issues with master

**Comment by Arsnael** [comment]
```
[ERROR] src/main/java/org/apache/james/imap/encode/base/ImapResponseComposerImpl.java:[29] (imports) CustomImportOrder: Wrong lexicographical order for 'org.apache.james.imap.api.Tag' import. Should be before 'org.apache.james.imap.api.display.ModifiedUtf7'.
```
Fixing it...

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
`Caused by: com.rabbitmq.client.ShutdownSignalException: channel error; protocol method: #method<channel.close>(reply-code=403, reply-text=ACCESS_REFUSED - queue 'taskManagerWorkQueue' in vhost '/' in exclusive use, class-id=60, method-id=20)`

test this please

**Comment by chibenwa** [comment]
```
[ERROR] Failed to execute goal on project james-server-lifecycle-api: Could not resolve dependencies for project org.apache.james:james-server-lifecycle-api:jar:3.5.0-SNAPSHOT: Failed to collect dependencies at com.google.guava:guava:jar:25.1-jre: Failed to read artifact descriptor for com.google.guava:guava:jar:25.1-jre: Could not transfer artifact com.google.guava:guava:pom:25.1-jre from/to central (https://repo.maven.apache.org/maven2): /root/.m2/repository/com/google/guava/guava/25.1-jre/guava-25.1-jre.pom.part (No such file or directory) -> [Help 1]
```

test this please

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**